controllers/imagerepository_controller.go

```diff
@@ -91,10 +95,20 @@ func (r *ImageRepositoryReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 			log.Error(err, "unable to update status")
 			return ctrl.Result{Requeue: true}, err
 		}
+		r.recordReadiness(imageRepo)
 		log.Info(msg)
 		return ctrl.Result{}, nil
 	}
 
+	// record reconciliation duration
+	if r.MetricsRecorder != nil {
+		objRef, err := reference.GetReference(r.Scheme, &imageRepo)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		defer r.MetricsRecorder.RecordDuration(*objRef, reconcileStart)
```

Comment:
Could the `recordReadiness` also be deferred -- it appears to be just before every exit.